src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -203,10 +257,43 @@ BuildSelectStatement(Query *masterQuery, List *masterTargetList, CustomScan *rem
 		topLevelPlan = &remoteScan->scan.plan;
 	}
 
+	/*
+	 * distinct on() requires sort + unique plans. If already has order by
+	 * clause we append distinct clauses to the end of it. Postgresql requires
+	 * that if both distinct on() and order by exists, ordering is first done
```

Comment:
This says distinct ordering is done first, but we add it to the end. Isn't this a contradiction, or am I missing something?

Reply:
I think I was not clear there.
I did not mean time ordering there. Requirement is that, if there is a distinct on clause, and order by clause. Order by clause MUST contain distinct on clauses in the beginning. Otherwise it would give an error.

Changed the comment a bit.